python/ql/lib/semmle/python/dataflow/new/Regexp.qll

```diff
@@ -25,18 +26,18 @@ deprecated module RegExpPatterns {
  * as a part of a regular expression.
  */
 class RegExpPatternSource extends DataFlow::CfgNode {
-  private Regex astNode;
+  private DataFlow::Node sink;
 
-  RegExpPatternSource() { astNode = this.asExpr() }
+  RegExpPatternSource() { this = regExpSource(sink) }
```

Comment:
I am tempted to turn the predicate `RegExpSource::regSink` into the charpred of a class, just so I could annotate the result type of `RexExpSource::regExpSource` with it. Then we could see it here as:
```
  private RegExpSink sink;

  RegExpPatternSource() { this = regExpSource(sink) }
```
(where I imagine the class was called `RegExpSink`). I think this might clarify the constraints on `sink` a little bit.